lightning/src/ln/channelmanager.rs

```diff
@@ -3131,6 +3157,80 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
 		Ok(())
 	}
 
+	/// Attempts to forward an intercepted HTLC over the provided scid and with the provided
+	/// amount to forward. Should only be called in response to an [`HTLCIntercepted`] event.
+	///
+	/// Intercepted HTLCs can be useful for Lightning Service Providers (LSPs) to open a just-in-time
+	/// channel to a receiving node if the node lacks sufficient inbound liquidity.
+	///
+	/// To make use of intercepted HTLCs, use [`ChannelManager::get_intercept_scid`] to generate short
+	/// channel id(s) to put in the receiver's invoice route hints. These route hints will signal to
+	/// LDK to generate an [`HTLCIntercepted`] event when it receives the forwarded HTLC, and this
+	/// method or [`ChannelManager::fail_intercepted_htlc`] MUST be called in response to the event.
```

Comment:
Responding to https://github.com/lightningdevkit/rust-lightning/pull/1601#discussion_r1015969808 here to keep the conversation within this PR :).

I think the wording "MUST" is good if we decide on making possibility of events/enums generation optin with a config flag, or default to intercepting HTLCs in the future :).

If we don't make this opt in now, I think MUST is a bit strong. As we won't require users to implement support for generating `Intercept` scids, I think it's a bit strange to word it as if we to require them to implement support for failing/forwarding `Intercepted` HTLCs, as they currently can't avoid that there's a possibility that a `HTLCIntercepted` event is generated. If we decide on keeping the possibility of events/enums generation not optin, perhaps wording it as "SHOULD" would work better?